src/rp2_common/pico_cyw43_arch/cyw43_arch_freertos.c

```diff
@@ -123,9 +123,13 @@ int cyw43_arch_init(void) {
     cyw43_worker_ran_sem = xSemaphoreCreateBinary();
 
 #if CYW43_LWIP
-    SemaphoreHandle_t init_sem = xSemaphoreCreateBinary();
-    tcpip_init(tcpip_init_done, init_sem);
-    xSemaphoreTake(init_sem, portMAX_DELAY);
+    static bool tcpip_init_called = false;
+    if (!tcpip_init_called) {
+      tcpip_init_called = true;
```

Comment:
Does access to this variable need to be protected by some kind of mutex, to prevent race conditions? :shrug:

Reply:
Ideally `cyw43_arch_init()` should only be called from one location, so I don't think a mutex is required here.